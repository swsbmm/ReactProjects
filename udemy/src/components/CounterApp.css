*{
    margin: 0;
    padding: 0;
    box-sizing: none;
}

.container {
    background-color: #4b7875;
    width: 100vw;
    height: 100vh;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
}

p{
    display: flex;
    justify-content: center;
    font-size: 50px;
    font-weight: bold;
}

button{
    padding: 2px;
    background-color: black;
    color: white;
    cursor: pointer;
    margin-right: 5px;
    height: 3rem;
    width: 5rem;
}