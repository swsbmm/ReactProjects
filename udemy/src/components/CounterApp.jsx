import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
// import './CounterApp.css';

export default function CounterApp({ value }) {
  const [count, setCount] = useState(value);
  const sumar = (event) => {
    setCount(count + 1);
  };
  const restar = () => setCount(count - 1);
  const reset = () => setCount(value);

  return (
    <Fragment>
      <div className="container">
        <h1>Counter App</h1>
        <div>
          <p>{count}</p>
          <div>
            <button onClick={sumar}>+1</button>
            <button onClick={restar}>-1</button>
            <button onClick={reset}>reset</button>
          </div>
        </div>
      </div>
    </Fragment>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
