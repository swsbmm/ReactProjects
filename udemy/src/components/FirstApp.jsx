import PropTypes from 'prop-types';

export const FirstApp = ( { title, subTitle, name }) => {


    return(
        <>
            <h1 data-testid="test-title">{ title }</h1>
            <p data-testid="test-subtitle">{ subTitle }</p>
            <p data-testid="test-subtitle">{ subTitle }</p>
            <p>{ name }-</p>
        </>

    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
    name: PropTypes.string,
}