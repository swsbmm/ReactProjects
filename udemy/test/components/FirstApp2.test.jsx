import { render, screen } from "@testing-library/react";
import { FirstApp } from '../../src/components/FirstApp';

describe('test in the fist component <FistApp/> ', () => { 

    const title = 'Hola Mundo'
    const subTitle = 'Este es el subtitulo.'

    test('debe hacer mach con el snapshot', () => {
        const { container } = render(<FirstApp title={ title }/>);
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar el mensaje "Hola Mundo"', ()=> {
        render(<FirstApp title={ title }/>);
        // screen.debug();u
        expect(screen.getByText(title)).toBeTruthy();
    })

    test('debe de mostrar el titulo de un h1', ()=> {
        render(<FirstApp title={title}/>);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    })

    test('debe mostrar el subtitulo enviado por las props', ()=>{
        render(<FirstApp title={title} subTitle={subTitle} />);
        expect(screen.getAllByText(subTitle).length).toBe(2);
    })

 })