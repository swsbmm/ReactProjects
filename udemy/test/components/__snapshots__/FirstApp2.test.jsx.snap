// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in the fist component <FistApp/>  debe hacer mach con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola Mundo
  </h1>
  <p
    data-testid="test-subtitle"
  />
  <p
    data-testid="test-subtitle"
  />
  <p>
    -
  </p>
</div>
`;
